src/PrestaShopBundle/Translation/Extractor/ThemeExtractor.php

```diff
@@ -93,6 +95,35 @@ public function extract(Theme $theme, $locale = 'en-US', $rootDir = false)
         throw new \LogicException(sprintf('The format %s is not supported.', $this->format));
     }
 
+    private function overrideFromDefaultCatalog($locale, &$catalogue)
+    {
+        $defaultCatalogue = $this->themeProvider
+            ->setLocale($locale)
+            ->getDefaultCatalogue()
+        ;
+
+        if (!empty($defaultCatalogue)) {
+            $defaultCatalogue = current($defaultCatalogue);
+
+            if (!empty($defaultCatalogue)) {
+                $defaultDomainsCatalogue = $catalogue->getDomains();
+
+                foreach ($defaultCatalogue as $domain => $translation) {
+                    $domain = str_replace('.' . $locale, '', $domain);
+                    $domain = implode('.', preg_split('/(?=[A-Z])/', $domain, -1, PREG_SPLIT_NO_EMPTY));
```

Comment:
Please add a comment explaining what you're doing. It took me a while to figure out you were splitting camelCased words 😛
Thanks!